File: features/document/src/main/java/com/nivasafinance/features/document/repository/DocumentRepository.java
package com.nivasafinance.features.document.repository;

import com.nivasafinance.features.document.entity.Document;
import org.javers.spring.annotation.JaversSpringDataAuditable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
@JaversSpringDataAuditable
public interface DocumentRepository extends JpaRepository<Document, Long> {
    Optional<Document> findByIdentifier(UUID identifier);
}

File: features/document/src/main/java/com/nivasafinance/features/document/repository/DocumentRepository.kt
package com.nivasafinance.features.document.repository

import com.nivasafinance.features.document.entity.Document
import org.javers.spring.annotation.JaversSpringDataAuditable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
@JaversSpringDataAuditable
interface DocumentRepository : JpaRepository<Document, UUID>
